lnd_test.go

```diff
@@ -1464,6 +1468,150 @@ func testMultiHopPayments(net *networkHarness, t *harnessTest) {
 	}
 }
 
+func testMaxFee(net *networkHarness, t *harnessTest) {
+	type testCase struct {
+		maxFee        int64
+		expectSuccess bool
+		hash          []byte
+	}
+	testCases := []testCase{
+		{
+			// maxFee < 0
+			maxFee:        -1,
+			expectSuccess: false,
+		},
+		{
+			// maxFee < TotalFees
+			maxFee:        0,
+			expectSuccess: false,
+		},
+		{
+			// maxFee == TotalFees
+			maxFee:        1001,
+			expectSuccess: true,
+		},
+		{
+			// maxFee > TotalFees
+			maxFee:        2000,
+			expectSuccess: true,
+		},
+	}
+	const chanAmt = btcutil.Amount(100000)
+	ctxb := context.Background()
+	timeout := time.Duration(time.Second * 5)
+
+	// Open a channel with 100k satoshis between Alice and Bob with Alice
+	// being the sole funder of the channel.
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
+	chanPointAlice := openChannelAndAssert(ctxt, t, net, net.Alice,
+		net.Bob, chanAmt, 0)
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err := net.Alice.WaitForNetworkChannelOpen(ctxt, chanPointAlice)
+	if err != nil {
+		t.Fatalf("alice didn't advertise her channel: %v", err)
+	}
+
+	// Create a new node (Carol), load her with some funds, then establish
+	// a connection between Carol and Alice with a channel that has
+	// identical capacity to the one created above.
+	//
+	// The network topology should now look like: Carol -> Alice -> Bob
+	carol, err := net.NewNode(nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, carol, net.Alice); err != nil {
+		t.Fatalf("unable to connect carol to alice: %v", err)
+	}
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, carol)
+	if err != nil {
+		t.Fatalf("unable to send coins to carol: %v", err)
+	}
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	chanPointCarol := openChannelAndAssert(ctxt, t, net, carol,
+		net.Alice, chanAmt, 0)
+
+	// Create invoices for Bob, which expect a payment from Carol for 1k
+	// satoshis with a different preimage each time.
+	const paymentAmt = 1000
+	for i := range testCases {
+		invoice := &lnrpc.Invoice{
+			Memo:  "testing",
+			Value: paymentAmt,
+		}
+		resp, err := net.Bob.AddInvoice(ctxb, invoice)
+		if err != nil {
+			t.Fatalf("unable to add invoice: %v", err)
+		}
+
+		testCases[i].hash = resp.RHash
+	}
+
+	// Wait for carol to recognize both the Channel from herself to Carol,
+	// and also the channel from Alice to Bob.
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = carol.WaitForNetworkChannelOpen(ctxt, chanPointCarol)
+	if err != nil {
+		t.Fatalf("carol didn't advertise her channel: %v", err)
+	}
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	err = carol.WaitForNetworkChannelOpen(ctxt, chanPointAlice)
+	if err != nil {
+		t.Fatalf("carol didn't see the alice->bob channel before timeout: %v", err)
+	}
+
+	// Using Carol as the source, pay to the 5 invoices from Bob created above.
```

Comment:
Comment doesn't match the code itself. Looks like it's been copy-pasta'd from elsewhere.